lib/src/models/data-item.dart

```diff
@@ -105,46 +104,63 @@ class DataItem implements TransactionBase {
   Future<Uint8List> getSignatureData() => deepHash(
         [
           utf8.encode('dataitem'),
-          utf8.encode('1'),
+          utf8.encode('1'), //Transaction format
+          utf8.encode('1'), //Signature type
           decodeBase64ToBytes(owner),
           decodeBase64ToBytes(target),
           decodeBase64ToBytes(nonce),
-          tags
-              .map(
-                (t) => [
-                  decodeBase64ToBytes(t.name),
-                  decodeBase64ToBytes(t.value),
-                ],
-              )
-              .toList(),
+          serializeTags(tags: tags),
           data,
         ],
       );
 
   /// Signs the [DataItem] using the specified wallet and sets the `id` and `signature` appropriately.
   @override
-  Future<void> sign(Wallet wallet) async {
+  Future<Uint8List> sign(Wallet wallet) async {
     final signatureData = await getSignatureData();
     final rawSignature = await wallet.sign(signatureData);
 
     _signature = encodeBytesToBase64(rawSignature);
 
     final idHash = await sha256.hash(rawSignature);
     _id = encodeBytesToBase64(idHash.bytes);
-  }
-
-  @override
-  Future<void> signWithRawSignature(Uint8List rawSignature) async {
-    _signature = encodeBytesToBase64(rawSignature);
-
-    final idHash = await sha256.hash(rawSignature);
-    _id = encodeBytesToBase64(idHash.bytes);
+    return Uint8List.fromList(idHash.bytes);
   }
 
   /// Verify that the [DataItem] is valid.
   @override
   Future<bool> verify() async {
+    final buffer = asBinary();
     try {
+      if (buffer.lengthInBytes < MIN_BINARY_SIZE) {
+        return false;
+      }
+      final sigType = byteArrayToLong(buffer.asUint8List().sublist(0, 2));
+      assert(sigType == 1);
+      var tagsStart = 2 + 512 + 512 + 2;
```

Comment:
Can we factor out all the magic numbers as constants to explain what they are based on the spec?

Reply:
[Created Ticket](https://ardrive.atlassian.net/browse/PE-715)